src/libs/ReportUtils.ts

```diff
@@ -1227,6 +1236,18 @@ function isIOURequest(report: OnyxEntry<Report>): boolean {
     return false;
 }
 
+/**
+ * A Track Expense Report is a thread where the parent the parentReportAction is a transaction, and
+ * parentReportAction has type of track.
+ */
+function isTrackExpenseReport(report: OnyxEntry<Report>): boolean {
+    if (isThread(report)) {
+        const parentReportAction = ReportActionsUtils.getParentReportAction(report);
+        return !isEmptyObject(parentReportAction) && ReportActionsUtils.isTrackExpenseAction(parentReportAction);
```

Comment:
Coming from #52240 BZ checklist: 

After update track expense to transaction, the type of parent action still is `track` so we have to check parentReportID with selfDMReportID to detect it's a track expense